internal/fwtype/static_collection_validation.go

```diff
@@ -140,3 +139,19 @@ func ReturnCollectionWithDynamicTypeDiag() diag.Diagnostic {
 			"If underlying dynamic values are required, replace the return definition with DynamicReturn instead.",
 	)
 }
+
+// IsAllowedPrimitiveType checks if the given attr.Type is an allowed primitive type for resource identity.
+func IsAllowedPrimitiveType(typ attr.Type) bool {
+	switch typ.(type) {
+	case basetypes.BoolType,
+		basetypes.Float64Type,
+		basetypes.Float32Type,
+		basetypes.Int64Type,
+		basetypes.Int32Type,
+		basetypes.NumberType,
+		basetypes.StringType:
+		return true
```

Comment:
I don't think it's super common, but I'm thinking we should use the "typable" interfaces here instead (which will give you a superset of the functionality you have here.

```suggestion
	case basetypes.BoolTypable,
		basetypes.Float64Typable,
		basetypes.Float32Typable,
		basetypes.Int64Typable,
		basetypes.Int32Typable,
		basetypes.NumberTypable,
		basetypes.StringTypable:
		return true
```

The difference there is that if you set the element type to a custom type that is a primitive (like an `iptypes.IPv4Address`, then it'll still recognize it as a valid list of strings)